slack-service/app.py

```diff
@@ -0,0 +1,182 @@
+import os
+from flask import Flask, request
+from slack_sdk import WebClient
+from asset_manager.agent_manager import AgentManager
+from llama_stack_client import LlamaStackClient
+
+app = Flask(__name__)
+
+# Initialize Slack Client
+slack_token = os.environ.get("SLACK_BOT_TOKEN")
+if not slack_token:
+    raise ValueError("SLACK_BOT_TOKEN environment variable not set!")
+slack_client = WebClient(token=slack_token)
+
+# Initialize Agent Manager
+agent_manager = None
+available_agents = {}
+user_sessions = {}
+
+
+def initialize_agent_manager():
+    """Initialize the agent manager and connect to LlamaStack"""
+    global agent_manager, available_agents
+
+    try:
+        llama_stack_host = os.environ.get("LLAMASTACK_SERVICE_HOST")
+        full_url = f"http://{llama_stack_host}:8321"
+        print(f"Connecting to: {full_url}")
+        # Create LlamaStack client
+        client = LlamaStackClient(base_url=full_url, timeout=600.0)
+
+        # Initialize agent manager
+        agent_manager = AgentManager({"timeout": 120})
+        agent_manager._client = client
+
+        # Get available agents
+        available_agents = agent_manager.agents()
+        print("Agent Manager initialized successfully")
+        print(f"   Available agents: {list(available_agents.keys())}")
+
+    except Exception as e:
+        print(f"Error initializing Agent Manager: {e}")
+        raise
+
+
+def send_message_to_agent(agent_id: str, session_id: str, messages: list) -> str:
+    """Send a message to an agent and return the response"""
+    try:
+        response_stream = agent_manager._client.agents.turn.create(
+            agent_id=agent_id,
+            session_id=session_id,
+            stream=True,
+            messages=messages,
+        )
+
+        response = ""
+        for chunk in response_stream:
+            # check if the chunk contains an error attribute before accessing it
+            if hasattr(chunk, "error") and chunk.error:
+                error_message = chunk.error.get("message", "Unknown agent error")
+                print(f"Error from agent API: {error_message}")
+                return f"Error from agent: {error_message}"
+
+            # If no error, process the stream for the final message
+            if (
+                hasattr(chunk, "event")
+                and hasattr(chunk.event, "payload")
+                and chunk.event.payload.event_type == "turn_complete"
+            ):
+                if hasattr(chunk.event.payload.turn, "output_message"):
+                    content = chunk.event.payload.turn.output_message.content
+                    response += content
+
+        return response.strip()
+
+    except Exception as e:
+        print(f"Error in send_message_to_agent: {e}")
+        return f"Error: {str(e)}"
+
+
+def handle_user_message(user_id: str, text: str, channel_id: str):
+    """Handle incoming user message and route to appropriate agent"""
+
+    # Create or get user session
```

Comment:
We probably want to pull this part of the code into a common code (maybe under a session_manager directory" so that we can share the code between the slack UI and the command line UI.